Description: Fix crash when the audio card has no mixers.
 Debian bugs:
  - http://bugs.debian.org/673858
  - http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=684363
Origin: upstream
Bug: https://code.google.com/p/volti/issues/detail?id=27
---
 src/alsactrl.py |    9 +++++++--
 src/mixer.py    |    7 ++++++-
 2 files changed, 13 insertions(+), 3 deletions(-)

--- volti.orig/src/alsactrl.py
+++ volti/src/alsactrl.py
@@ -45,7 +45,7 @@ class AlsaControl():
                 self.reopen(self.card_index, control)
                 self._check_version()
             except Exception, err:
-                log.Error("can't open first available control for card %s\nerror: %s" % (
+                log.Warn("can't open first available control for card %s\nerror: %s" % (
                     self.get_card_name(), str(err)))
 
     def __del__(self):
@@ -189,7 +189,12 @@ class AlsaControl():
     def get_mixers(self, card_index=0):
         """ Returns mixers list """
         mixers = []
-        for mixer in alsa.mixers(card_index):
+        amixers = []
+        try:
+            amixers = alsa.mixers(card_index)
+        except alsa.ALSAAudioError:
+            pass
+        for mixer in amixers:
             try:
                 m = alsa.Mixer(control=mixer, cardindex=card_index)
                 cap = m.volumecap()
--- volti.orig/src/mixer.py
+++ volti/src/mixer.py
@@ -139,7 +139,12 @@ class Mixer(gtk.Window):
                         self.cp.get("card-%d" % card_index, "mask_control"))
             except:
                 self.control_mask[card_index] = 0
-                for count, mixer in enumerate(alsa.mixers(card_index)):
+                amixers = []
+                try:
+                    amixers = alsa.mixers(card_index)
+                except alsa.ALSAAudioError:
+                    pass
+                for count, mixer in enumerate(amixers):
                     self.control_mask[card_index] |= (1 << count)
 
             n = 0
